sklearn/isotonic.py

```diff
@@ -292,7 +310,7 @@ def _build_y(self, X, y, sample_weight, trim_duplicates=True):
         # Handle the left and right bounds on X
         self.X_min_, self.X_max_ = np.min(X), np.max(X)
 
-        if trim_duplicates:
+        if trim_duplicates and not self.centered:
```

Comment:
Does the option trim_duplicates make sense to CIR? If yes the logic should be
```python
if trim_duplicates:
    keep_data = ...
    if self.centered:
        X, y = self._build_cir_points(X[keep_data], y[keep_data], sample_weight[keep_data])
        return X, y
    else:
        return X[keep_data], y[keep_data]
```
Of it does not make sense, it would be
```python
if self.centered:
    X, y = self._build_cir_points(X, y, sample_weight)
    return X, y
if trim_duplicates:
    # code unchanged
```

Reply:
Trimming duplicates in combination with CIR makes no sense. Updated according to your suggestion.